rules/cakephp/src/Rector/MethodCall/ArrayToFluentCallRector.php

```diff
@@ -0,0 +1,223 @@
+    {
+        return new RectorDefinition('Moves array options to fluent setter method calls.', [
+            new CodeSample(
+                <<<'PHP'
+class ArticlesTable extends \Cake\ORM\Table
+{
+    public function initialize(array $config)
+    {
+        $this->belongsTo('Authors', [
+            'foreignKey' => 'author_id',
+            'propertyName' => 'person'
+        ]);
+    }
+}
+PHP
+                ,
+                <<<'PHP'
+class ArticlesTable extends \Cake\ORM\Table
+{
+    public function initialize(array $config)
+    {
+        $this->belongsTo('Authors')
+            ->setForeignKey('author_id')
+            ->setProperty('person');
+    }
+}
+PHP
+            ),
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [MethodCall::class];
+    }
+
+    /**
+     * @param MethodCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $config = $this->matchTypeAndMethodName($node);
+        if ($config === null) {
+            return null;
+        }
+
+        [$argument, $class] = $config;
+
+        $fluentMethods = $this->resolveFluentMethodCalls($class);
+        if (! $fluentMethods) {
+            return null;
+        }
+
+        return $this->replaceArrayToFluentMethodCalls($node, $argument, $fluentMethods);
+    }
+
+    /**
+     * @return mixed[]|null
+     */
+    private function matchTypeAndMethodName(MethodCall $methodCall): ?array
+    {
+        foreach ($this->factoryMethods as $className => $methodName) {
+            if (! $this->isObjectType($methodCall->var, $className)) {
+                continue;
+            }
+
+            /** @var string[] $methodNames */
+            $methodNames = array_keys($methodName);
+            if (! $this->isNames($methodCall->name, $methodNames)) {
+                continue;
+            }
+
+            $currentMethodName = $this->getName($methodCall->name);
+            if ($currentMethodName === null) {
+                continue;
+            }
+
+            $config = $methodName[$currentMethodName];
+            $argument = $config[self::ARGUMENT] ?? 1;
+            $class = $config[self::CLASSNAME];
+
+            return [$argument, $class];
+        }
+
+        return null;
+    }
+
+    private function resolveFluentMethodCalls(string $class)
+    {
+        if (! isset($this->configurableClasses[$class])) {
+            return [];
+        }
+
+        $map = $this->configurableClasses[$class];
+
+        if (isset($map[self::EXTEND])) {
+            $parent = $map[self::EXTEND];
+            unset($map[self::EXTEND]);
+            $map = array_merge($this->resolveFluentMethodCalls($parent), $map);
+        }
+
+        return $map;
+    }
+
+    /**
+     * @param string[] $fluentMethods
+     */
+    private function replaceArrayToFluentMethodCalls(
+        MethodCall $methodCall,
+        int $argument,
```

Comment:
Is this argument count? Or argument position?

Reply:
Argument position.